### PR TITLE
Add Django 1.11 support to this project

### PR DESCRIPTION
**Pull request by chosak**
This commit adds Django 1.11 support to this project, which previously only supported Django 1.8. As part of adding that support, it also standardizes some pieces of this project to be more in line with Django and Bureau coding conventions.

Changes include:

- The use of `requirements.txt` has been deprecated to instead put Python dependencies in `setup.py`.
- The `README.md` file has been updated to reflect changes in testing and installation instructions.
- The `.coveragerc` file has been simplified to have a single `source` directive to cover the `retirement_api` folder instead of excluding a fixed set of folders.
- Previously when running in standalone mode the URLs for this app were hardcoded to assume a `/retirement/` prefix to the path, because that's how it is deployed in cfgov-refresh. This isn't an assumption that we should have to make when running an app standalone, and so I've removed
that prefix and made this app able to run more flexibly. This means removing a few places with hardcoded `/retirement/` URLs and instead relying on Django's URL reversing.
- Related to the above, this change now groups this app's view names and
templates under a `retirement_api` namespace to better isolate them from the rest of Django.
- Previously the link to the "about" page from the claiming page made use of Django's internationalization framework in a strange way. Instead of "translating" a URL to Spanish to get the proper about URL in that language, the claiming view now passes in the appropriate view name as
part of the context.
- The test setup has been modified in a few ways: tox is now used to test different versions of Django, the test settings use a more conventional `settings.TEMPLATES` to define templates (instead of using a fixed subdirectory), and test URLs have been split out into their own file.

To test, verify that these URLs work when running locally:

- http://localhost:8000/before-you-claim/

  Go through the entire flow, verify that the API calls work. Also
  verify that the "Where do these numbers come from?" link works properly
  to go to the About page.

- http://localhost:8000/before-you-claim/es/

  Same as above, in Spanish.

- http://localhost:8000/before-you-claim/about/

- http://localhost:8000/before-you-claim/about/es/

@anselmbradford does [this change](https://github.com/cfpb/retirement/compare/master...chosak:django-1.11?expand=1#diff-ff44548bd0cbe70dc467825ebc65143cL10) seem reasonable to you as a way to make the root path more flexible?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,15 +6,12 @@ env: DJ_KEY=testkey
 before_install:
 - nvm use 6
```
Comment (higs4281) [inline .travis.yml]: After @anselmbradford 's modernizations, I don't think we need to specify node 6 anymore. the frontend build works for me with node 8.

Comment (anselmbradford) [reply .travis.yml]: Yes!

**Comment by anselmbradford** [comment]
> does this change seem reasonable to you as a way to make the root path more flexible?

That seems fine. Alternatively you can use nodes path module https://nodejs.org/api/path.html#path_path_relative_from_to

File: .travis.yml (hunk 1 above)
Comment (chosak) [reply .travis.yml]: Thanks, addressed in cebb17d.

File: README.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ Tú puedes ver este app en español por poner `/es` al parte final del url.
  * [Gulp](http://gulpjs.com/)
```
[the remark this answers is not in the document]
Comment (chosak) [reply README.md]: Thanks, addressed in 926af5f.

[the remark this answers is not in the document]
Comment (anselmbradford) [reply README.md]: Bower is effectively [dead](https://bower.io/blog/2017/how-to-migrate-away-from-bower/), so we should strip it out completely in another PR.

**Pull request merged**